source/vibe/http/common.d

```diff
@@ -469,20 +488,25 @@ final class ChunkedOutputStream : OutputStream {
 	{
 		assert(!m_finalized);
 		if( m_buffer.data.length > 0 ) flush();
+		ubyte[] buffer = null;
+		auto getBytes = (size_t size) {
+			if (buffer is null || buffer.length < size)
+				buffer = new ubyte[](size);
+				return buffer[0..size];
+			};
+
 		if( nbytes == 0 ){
 			while( !data.empty ){
 				auto sz = data.leastSize;
 				assert(sz > 0);
-				writeChunkSize(sz);
-				m_out.write(data, sz);
-				m_out.write("\r\n");
-				m_out.flush();
+				auto bytes = getBytes(sz);
+				data.read(bytes);
+				write(bytes);
```

Comment:
Thank you. Actually, once this `put()` overload is implemented, one could even build something like `AllocAppender.put(InputStream data, ulong size)` on top of it - this seems (at least to me) to be a much more obvious use case from an outside perspective.